### PR TITLE
feat: add linting and building to helm-release

### PR DESCRIPTION
**Pull request by KFearsoff**
This PR should be backwards-compatible for all workflows called from `release.yaml` workflows (which is currently all of them). Some things of note:
- I've removed the branching at "Update versions" step. Since workflows are called from `release.yaml`, we already have a tag, so we'd never actually go to the `else` part
- I've used `working-directory` clause instead of specifying paths with environment variables. Environment variables are pretty unreliable and prone to error, might as well use the built-in solution
- If `ENVIRONMENT` is passed, then we'll use `values/$ENVIRONMENT.yaml` for `helm lint`; otherwise, `helm lint .` is used. Needed for backwards compatibility
- Actually updating the versions and committing them to the repo happens only if the workflow is ran on tag. This is what we currently do anyway

### CODE REVIEW

File: .github/workflows/helm-release.yaml [hunk 1]
```diff
@@ -13,32 +13,48 @@ on:
...
           else
-            echo "Updating $CHART/values/$ENVIRONMENT.yaml"
-            APP_VERSION="sha-$(echo ${GITHUB_SHA} | cut -c1-7)"
-            sed -i "s#^  tag:.*#  tag: ${APP_VERSION}#g" $CHART/values/$ENVIRONMENT.yaml
-            cat $CHART/values/$ENVIRONMENT.yaml | head -10
+            helm lint .
           fi
+
+      - name: Update versions
+        if: ${{ github.ref_type == 'tag' }}
```
Comment (DovnarAlexander) [inline .github/workflows/helm-release.yaml:37]: Should be executed every time, not only for tags.

File: .github/workflows/helm-release.yaml [hunk 2]
```diff
@@ -13,32 +13,48 @@ on:
...
+        env:
+          ENVIRONMENT: ${{ inputs.ENVIRONMENT }}
+        run: |
+          APP_VERSION="${GITHUB_REF_NAME##*/}"
+          echo "Updating Chart.yaml"
+          sed -i "s#^version:.*#version: ${APP_VERSION/v/}#g" Chart.yaml
+          sed -i "s#^appVersion:.*#appVersion: ${APP_VERSION}#g" Chart.yaml
+          cat Chart.yaml
+
       - name: Commit file to branch
```
Comment (DovnarAlexander) [inline .github/workflows/helm-release.yaml:48]: Should be the latest step dependent on previous ones

**Review by DovnarAlexander — CHANGES_REQUESTED** [review]

**Comment by KFearsoff** [comment]
Fixed

File: .github/workflows/helm-release.yaml (hunk 2 above)
Comment (DovnarAlexander) [reply .github/workflows/helm-release.yaml:51]: Should be `needs` block here to build direct dependency

File: .github/workflows/helm-release.yaml [hunk 3]
```diff
@@ -13,32 +13,47 @@ on:
...
+        run: |
+          APP_VERSION="${GITHUB_REF_NAME##*/}"
+          echo "Updating Chart.yaml"
+          sed -i "s#^version:.*#version: ${APP_VERSION/v/}#g" Chart.yaml
+          sed -i "s#^appVersion:.*#appVersion: ${APP_VERSION}#g" Chart.yaml
+          cat Chart.yaml
+
+      - name: Build Helm chart
+        working-directory: ${{ inputs.CHART_LOCATION }}
+        run: helm package .
```
Comment (DovnarAlexander) [inline .github/workflows/helm-release.yaml:49]: Should be `needs` block here to build direct dependency to do that step only after `Update versions` and `Lint`

**Review by DovnarAlexander — CHANGES_REQUESTED** [review]

**Review by DovnarAlexander — APPROVED** [review]

**Pull request merged**